Source/Core/GroundPolylineGeometry.js

```diff
@@ -0,0 +1,1054 @@
+
+    var MITER_BREAK_SMALL = Math.cos(CesiumMath.toRadians(1));
+    var MITER_BREAK_LARGE = Math.cos(CesiumMath.toRadians(179));
+
+    // Initial heights for constructing the wall.
+    // Keeping WALL_INITIAL_MIN_HEIGHT near the ellipsoid surface helps
+    // prevent precision problems with planes in the shader.
+    // Putting the start point of a plane at ApproximateTerrainHeights._defaultMinTerrainHeight,
+    // which is a highly conservative bound, usually puts the plane origin several thousands
+    // of meters away from the actual terrain, causing floating point problems when checking
+    // fragments on terrain against the plane.
+    // Ellipsoid height is generally much closer.
+    // The initial max height is arbitrary.
+    // Both heights are corrected using ApproximateTerrainHeights for computing the actual volume geometry.
+    var WALL_INITIAL_MIN_HEIGHT = 0.0;
+    var WALL_INITIAL_MAX_HEIGHT = 1000.0;
+
+    /**
+     * A description of a polyline on terrain. Only to be used with GroundPolylinePrimitive.
+     *
+     * @alias GroundPolylineGeometry
+     * @constructor
+     *
+     * @param {Object} options Options with the following properties:
+     * @param {Number} [options.width=1.0] The screen space width in pixels.
+     * @param {Cartesian3[]} options.positions An array of {@link Cartesian3} defining the polyline's points. Heights above the ellipsoid will be ignored.
+     * @param {Number} [options.granularity=9999.0] The distance interval used for interpolating options.points. Defaults to 9999.0 meters. Zero indicates no interpolation.
+     * @param {Boolean} [options.loop=false] Whether during geometry creation a line segment will be added between the last and first line positions to make this Polyline a loop.
+     * @param {Ellipsoid} [options.ellipsoid=Ellipsoid.WGS84] Ellipsoid that input positions will be clamped to.
+     * @param {MapProjection} [options.projection] Map Projection for projecting coordinates to 2D.
+     *
+     * @exception {DeveloperError} At least two positions are required.
+     *
+     * @see GroundPolylineGeometry#createGeometry
+     * @see GroundPolylinePrimitive
+     */
+    function GroundPolylineGeometry(options) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+        var positions = options.positions;
+
+        //>>includeStart('debug', pragmas.debug);
+        if ((!defined(positions)) || (positions.length < 2)) {
+            throw new DeveloperError('At least two positions are required.');
+        }
+        //>>includeEnd('debug');
+
+        /**
+         * The screen space width in pixels.
+         * @type {Number}
+         */
+        this.width = defaultValue(options.width, 1.0); // Doesn't get packed, not necessary for computing geometry.
+
+        this._positions = positions;
+
+        /**
+         * The distance interval used for interpolating options.points. Zero indicates no interpolation.
```

Comment:
I picked `9999.0` as the default since it seems like this is the largest value that that permits two decimal places with 32-bit floating point.

I don't know that users would want to use polylines on terrain for millimeter-accuracy applications, especially since we try to maintain a constant screen-space width.
These also won't work on 3D Tilesets yet, and getting that to work robustly is going to be really tricky to scope out.

Reply:
> I picked 9999.0 as the default since it seems like this is the largest value that that permits two decimal places with 32-bit floating point.

Think of a user clicking out centerlines in a road, just make sure you are happy with the precision for that.

> These also won't work on 3D Tilesets yet, and getting that to work robustly is going to be really tricky to scope out.

Why?  The height of the plane/box?  Or that 3D Tiles is not 2.5D?  If there isn't already, please start a roadmap issue for brainstorming.

Reply:
> Think of a user clicking out centerlines in a road, just make sure you are happy with the precision for that.

The problem is the simulated constant screen space width, it makes using this for anything requiring millimeter precision very dicey. It should be okay though.

> These also won't work on 3D Tilesets yet, and getting that to work robustly is going to be really tricky to scope out.

In addition to 3D vs 2.5D and shadow volume height, not having tileset depth information available in classification passes is a big thing.

I'll add more notes here: https://github.com/AnalyticalGraphicsInc/cesium/issues/6636